### PR TITLE
redo(ticdc): set default timeout for accessing external storage to 15 minutes

### LINKED ISSUE #8089: Redo log can only tolerate storage error for 6s
### What did you do?

1. Create CDC changefeed with redo log
2. Run workload
3. Inject network partition to redo log storage for 10min

### What did you expect to see?

CDC should be able to tolerate 10min for redo log storage access error

### What did you see instead?

CDC currently only tolerates storage error for 6s: 

img_v2_cb9f0ce1-2ad3-4092-af60-69d6e3ca6c3g

### Versions of the cluster

bash-5.1# /cdc version
Release Version: v6.5.0-11-g3808d9732-dirty
Git Commit Hash: 3808d9732201b004147628abf4914fd2c5dd0711
Git Branch: release-6.5
UTC Build Time: 2023-01-05 07:30:18
Go Version: go version go1.19.1 linux/amd64
Failpoint Build: false

### PR DESCRIPTION
**Pull request by CharlesCheung96**
### What problem does this PR solve?

Issue Number: close #8089 

### What is changed and how it works?
Set the default timeout for accessing external storage to 15 minutes

#### Tests 

 - Unit test
 - Integration test
 - Manual test 
 1. run consistent_replicate_storage_s3 test
 2. stop minio manually
 3. sleep 5 minutes
 4. restart minio
 The test passed and the following logs were found:
image

### Release note 

```release-note
`Fix the issue that redo log cannot tolerate network faults for more than 6 seconds`.
```

### CODE REVIEW

File: pkg/redo/config.go [hunk 1]
```diff
@@ -155,13 +162,21 @@ func IsBlackholeStorage(scheme string) bool {
...
 // InitExternalStorage init an external storage.
 var InitExternalStorage = func(ctx context.Context, uri url.URL) (storage.ExternalStorage, error) {
+	s, err := util.GetExternalStorageWithTimeout(ctx, uri.String(), DefaultTimeout)
+	if err != nil {
+		return nil, errors.WrapChangefeedUnretryableErr(errors.ErrStorageInitialize, err)
+	}
+	return s, nil
+}
+
+func initExternalStorageForTest(ctx context.Context, uri url.URL) (storage.ExternalStorage, error) {
```
Comment (zhaoxinyu) [inline pkg/redo/config.go:172]: There is no test file which uses `initExternalStorageForTest`, maybe `initExternalStorage` is enough?

File: pkg/util/external_storage.go [hunk 2]
```diff
@@ -90,3 +112,102 @@ func DefaultS3Retryer() request.Retryer {
...
+	ctx context.Context, oldFileName, newFileName string,
+) error {
+	ctx, cancel := context.WithTimeout(ctx, s.timeout)
+	defer cancel()
+	return s.ExternalStorage.Rename(ctx, oldFileName, newFileName)
+}
+
+// IsNotExistInExtStorage checks if the error is caused by the file not exist in external storage.
+func IsNotExistInExtStorage(err error) bool {
+	if err != nil {
```
Comment (zhaoxinyu) [inline pkg/util/external_storage.go:188]: We can use the following check to reduce the number of nested levels:
```
if err == nil {
    return false
}
```

File: pkg/redo/config.go (hunk 1 above)
Comment (CharlesCheung96) [reply pkg/redo/config.go:172]: [ValidateStorage](https://github.com/pingcap/tiflow/pull/8181/files#diff-c39f6b418c48bd2b7281aa46aa529d775dffc376505422d2e4ebff54630da0b7R187) use this method to test external storage when create changefeed.

**Review by asddongmen — APPROVED** [review]

**Review by nongfushanquan — APPROVED** [review]

**Pull request merged**